Realm/RLMObservation.mm

```diff
@@ -236,6 +244,18 @@ auto reverse(Container const& c) {
         return value;
     }
 
+    if (lastProp.set) {
```

Comment:
This is the exact same code as the array case and could just be combined with that.